### PR TITLE
Added recipe for `py7zr`

### PR DESCRIPTION
**Pull request by sarthakpati**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by sarthakpati** [comment]
@conda-forge/help-python this is ready for review!

The failing Windows builds are related to https://github.com/conda-forge/pyppmd-feedstock/pull/1

File: recipes/py7zr/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 0ab402f73fc8cc41f2a5523436ae53dec6ff612597d9168ed80cfa1574d27fe0
+
+build:
+  number: 0
```
Comment (dopplershift) [inline recipes/py7zr/meta.yaml:14]: If this is pure Python can't this be `noarch: python`?
```suggestion
  number: 0
  noarch: python
```

File: recipes/py7zr/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  skip: true   # [py2k]
```
Comment (dopplershift) [inline recipes/py7zr/meta.yaml:15]: Not necessary, we're not building Python 2 packages any more.
```suggestion
```

File: recipes/py7zr/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    # The tarball shouldn't have egg-info--this one has paths in SOURCES.txt
+    # that break the build on windows.
+    - rm -rf py7zr.egg-info  # [not win]
+    - del py7zr.egg-info  # [win]
+    - {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - pip
+    - python
```
Comment (dopplershift) [inline recipes/py7zr/meta.yaml:28]: For noarch:
```suggestion
    - python >=3.6
```

File: recipes/py7zr/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    - setuptools-scm >=6.0.1
+  run:
+    - bcj-cffi >=0.5.1,<0.6.0
+    - brotli-python >=1.0.9
+    - brotlicffi >=1.0.9.2
+    - importlib_metadata
+    - multivolumefile >=0.2.3
+    - pycryptodomex >=3.6.6
+    - pyppmd >=0.15.0
+    - python
```
Comment (dopplershift) [inline recipes/py7zr/meta.yaml:39]: Again noarch:
```suggestion
    - python >=3.6
```

File: recipes/py7zr/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  commands:
+    - pip check
+    - py7zr --help
+  requires:
+    - pip
+
+about:
+  home: https://github.com/miurahr/py7zr
+  summary: Pure python 7-zip library
+  license: LGPL-2.1
```
Comment (dopplershift) [inline recipes/py7zr/meta.yaml:55]:
```suggestion
  license: LGPL-2.1-or-later
```

**Comment by sarthakpati** [comment]
@dopplershift this is ready for review! 😄

**Comment by sarthakpati** [comment]
FYI, this recipe is needed for [another](https://github.com/sarthakpati/staged-recipes/tree/pycox).

**Review by dopplershift — APPROVED** [review]

**Pull request merged**